sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Verifier.cs

```diff
@@ -0,0 +1,679 @@
+                compilationOptions = compilationOptions.WithSpecificDiagnosticOptions(
+                    supportedDiagnostics
+                        .Select(diagnostic =>
+                            new KeyValuePair<string, ReportDiagnostic>(diagnostic.Id, ReportDiagnostic.Warn))
+                        .Union(
+                            new[]
+                            {
+                                new KeyValuePair<string, ReportDiagnostic>(AnalyzerFailedDiagnosticId, ReportDiagnostic.Error)
+                            }));
+
+                var compilationWithOptions = compilation.WithOptions(compilationOptions);
+                var compilationWithAnalyzer = compilationWithOptions
+                    .WithAnalyzers(
+                        diagnosticAnalyzers.ToImmutableArray(),
+                        cancellationToken: tokenSource.Token);
+
+                return compilationWithAnalyzer.GetAllDiagnosticsAsync().Result;
+            }
+        }
+
+        #endregion
+
+        #region Codefix helper
+
+        private static void RunCodeFixWhileDocumentChanges(DiagnosticAnalyzer diagnosticAnalyzer, CodeFixProvider codeFixProvider,
+            string codeFixTitle, Document document, ParseOptions parseOption, string pathToExpected)
+        {
+            var currentDocument = document;
+            CalculateDiagnosticsAndCode(diagnosticAnalyzer, currentDocument, parseOption, out var diagnostics, out var actualCode);
+
+            diagnostics.Should().NotBeEmpty();
+
+            string codeBeforeFix;
+            var codeFixExecutedAtLeastOnce = false;
+
+            do
+            {
+                codeBeforeFix = actualCode;
+
+                var codeFixExecuted = false;
+                for (var diagnosticIndexToFix = 0; !codeFixExecuted && diagnosticIndexToFix < diagnostics.Count; diagnosticIndexToFix++)
+                {
+                    var codeActionsForDiagnostic = GetCodeActionsForDiagnostic(codeFixProvider, currentDocument, diagnostics[diagnosticIndexToFix]);
+
+                    if (TryGetCodeActionToApply(codeFixTitle, codeActionsForDiagnostic, out var codeActionToExecute))
+                    {
+                        currentDocument = ApplyCodeFix(currentDocument, codeActionToExecute);
+                        CalculateDiagnosticsAndCode(diagnosticAnalyzer, currentDocument, parseOption, out diagnostics, out actualCode);
+
+                        codeFixExecutedAtLeastOnce = true;
+                        codeFixExecuted = true;
+                    }
+                }
+            } while (codeBeforeFix != actualCode);
+
+            codeFixExecutedAtLeastOnce.Should().BeTrue();
+
+            var actualWithUnixLineEnding = actualCode.Replace(WindowsLineEnding, UnixLineEnding);
+            var expectedWithUnixLineEnding = File.ReadAllText(pathToExpected).Replace(WindowsLineEnding, UnixLineEnding);
+            actualWithUnixLineEnding.Should().Be(expectedWithUnixLineEnding);
```

Comment:
Same here I force unix line ending on both actual and expected. Note that we still need to do it on the `actualCode` because the fix can provide other line returns (for example header fix).